File: api/commandes/apiCommandes/src/main/java/fr/univamu/iut/apicommandes/Produit.java
package fr.univamu.iut.apicommandes;

public class Produit {
    private int idProduit;
    private String nomProduit;
    private int quantite;
    private String unite;

    // Constructor with all attributes
    public Produit(int idProduit, String nomProduit, int quantite, String unite) {
        this.idProduit = idProduit;
        this.nomProduit = nomProduit;
        this.quantite = quantite;
        this.unite = unite;
    }

    // Default constructor
    public Produit() {
    }

    // Getters and Setters
    public int getIdProduit() {
        return idProduit;
    }

    public void setIdProduit(int idProduit) {
        this.idProduit = idProduit;
    }

    public String getNomProduit() {
        return nomProduit;
    }

    public void setNomProduit(String nomProduit) {
        this.nomProduit = nomProduit;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public String getUnite() {
        return unite;
    }

    public void setUnite(String unite) {
        this.unite = unite;
    }

    // toString method
    @Override
    public String toString() {
        return "Produit{" +
                "idProduit=" + idProduit +
                ", nomProduit='" + nomProduit + '\'' +
                ", quantite=" + quantite +
                ", unite='" + unite + '\'' +
                '}';
    }
}

File: ihm/domain/Produit.php
<?php

namespace domain;

class Produit
{
    protected $idProduit;
    protected $nomProduit;
    protected $quantite;
    protected $unite;

    public function __construct($idProduit, $nomProduit, $quantite, $unite)
    {
        $this->idProduit = $idProduit;
        $this->nomProduit = $nomProduit;
        $this->quantite = $quantite;
        $this->unite = $unite;
    }

    public function getIdProduit()
    {
        return $this->idProduit;
    }

    public function getNomProduit()
    {
        return $this->nomProduit;
    }

    public function getQuantite()
    {
        return $this->quantite;
    }

    public function getUnite()
    {
        return $this->unite;
    }
}
